server/src/main/java/com/ncc/neon/better/PreprocessorNlpModule.java

```diff
@@ -42,13 +48,12 @@ protected void initEndpoints(HttpEndpoint[] endpoints) {
         }
     }
 
-    public Mono<RestStatus> performPreprocessing(String filename) {
+    public Mono<?> performPreprocessing(String filename) {
         HashMap<String, String> params = new HashMap<>();
         params.put("file", filename);
         return this.performListOperation(filename, listEndpoint)
                 .flatMap(pendingFiles -> this.initPendingFiles(pendingFiles)
                 .then(this.performNlpOperation(params, preprocessEndpoint)
-                .doOnError(onError -> this.handleNlpOperationError((WebClientResponseException) onError, pendingFiles)))
-                .flatMap(this::handleNlpOperationSuccess));
```

Comment:
So it doesn't call `handleNlpOperationSuccess` any more? That seems like a big change in the contract of the class.

Reply:
Ah this change was not intended

Reply:
This change was from April from commit 3c389267fee702c9200635a3bcaa73c7e616d3ac
Seems like it's related to adding in support for canceling operations.